.github/workflows/ci.yml

```diff
@@ -0,0 +1,57 @@
+name: Continuous Integration
+on: [push, pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    services:
+      postgres:
+        image: postgres:11
+        env:
+          POSTGRES_PASSWORD: postgres
+        ports: ['5432:5432']
```

Comment:
Nit: Is there a particular reason to specify the port? The [sample](https://docs.github.com/en/actions/using-containerized-services/creating-postgresql-service-containers) doesn't. Likewise for Redis.

Reply:
Not sure but it doesn't work without it: https://github.com/NZOI/nztrain/actions/runs/7363696558/job/20043401850

Reply:
I misunderstood the docs, what we are doing corresponds to "Running jobs directly on the runner machine" which *does* require port forwarding.